cmd/installer/cmd_deploy.go

```diff
@@ -0,0 +1,236 @@
+		if len(validationErrors) > 0 {
+			logger.Error("⛔ The provided configuration files are invalid:")
+
+			for _, e := range validationErrors {
+				subLogger.Errorf("%v", e)
+			}
+
+			return errors.New("please review your configuration and try again")
+		}
+
+		logger.Info("✅ Provided configuration is valid.")
+
+		// prepapre Kubernetes and Helm clients
+		kubeconfig := ctx.String("kubeconfig")
+		if len(kubeconfig) == 0 {
+			return errors.New("no kubeconfig (--kubeconfig or $KUBECONFIG) given")
+		}
+
+		kubeContext := ctx.String("kube-context")
+		helmTimeout := ctx.Duration("helm-timeout")
+
+		ctrlConfig, err := ctrlruntimeconfig.GetConfigWithContext(kubeContext)
+		if err != nil {
+			return fmt.Errorf("failed to get config: %v", err)
+		}
+
+		mgr, err := manager.New(ctrlConfig, manager.Options{})
+		if err != nil {
+			return fmt.Errorf("failed to construct mgr: %v", err)
+		}
+
+		// start the manager in its own goroutine
+		go func() {
+			if err := mgr.Start(wait.NeverStop); err != nil {
+				logger.Fatalf("Failed to start Kubernetes client manager: %v", err)
+			}
+		}()
+
+		appContext := context.Background()
+
+		// wait for caches to be synced
+		mgrSyncCtx, cancel := context.WithTimeout(appContext, 30*time.Second)
+		defer cancel()
+		if synced := mgr.GetCache().WaitForCacheSync(mgrSyncCtx.Done()); !synced {
+			logger.Fatal("Timed out while waiting for Kubernetes client caches to synchronize.")
+		}
+
+		kubeClient := mgr.GetClient()
+
+		if err := apiextensionsv1beta1.AddToScheme(mgr.GetScheme()); err != nil {
+			return fmt.Errorf("failed to add scheme: %v", err)
+		}
+
+		if err := operatorv1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
+			return fmt.Errorf("failed to add scheme: %v", err)
+		}
+
+		helmClient, err := helm.NewCLI(kubeconfig, kubeContext, helmTimeout, logger)
+		if err != nil {
+			return fmt.Errorf("failed to create Helm client: %v", err)
+		}
+
+		logger.Info("🧩 Deploying kubermatic stack…")
+		opt := kubermatic.Options{
+			HelmValues:                 helmValues,
+			KubermaticConfiguration:    kubermaticConfig,
+			RawKubermaticConfiguration: rawKubermaticConfig,
+			ForceHelmReleaseUpgrade:    ctx.Bool("force"),
+		}
+
+		if err := kubermatic.Deploy(appContext, subLogger, kubeClient, helmClient, opt); err != nil {
+			return err
+		}
+
+		logger.Infof("🛬 Installation completed successfully. %s", greeting())
+
+		return nil
+	}))
+}
+
+func loadKubermaticConfiguration(filename string) (*operatorv1alpha1.KubermaticConfiguration, *unstructured.Unstructured, error) {
+	if filename == "" {
+		return nil, nil, errors.New("no file specified via --config flag")
+	}
+
+	content, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	raw := &unstructured.Unstructured{}
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(content), 1024).Decode(raw); err != nil {
+		return nil, nil, fmt.Errorf("failed to decode %s: %v", filename, err)
+	}
```

Comment:
Need the raw config in order to not accidentally created lots of empty objects in the cluster. Loading the config and unmarshalling it directly into a KubermaticConfiguration would interfere with the defaulting the operator does later.